app/src/main/java/com/base12innovations/android/fireroad/models/course/CourseSearchEngine.java

```diff
@@ -172,7 +172,7 @@ public static void exactMatch(EnumSet<Filter> filter) {
             else {
                 for (String comp : queryComps) {
                     boolean found = false;
-                    String regex = filters.contains(Filter.MATCHES) ? "^.*[^A-z0-9]" + Pattern.quote(comp) + "[^A-z0-9].*$" : "";
+                    String regex = filters.contains(Filter.MATCHES) ? "^.*" + Pattern.quote(comp) + ".*$" : "";
```

Comment:
Is this change necessary? It should be responsible for doing an exact (full-word) match as opposed to "contains".

Reply:
I see, it just happens that the field sometimes just has the word/course itself (just "6.009") so the character requirement rejects it. It should be fixed now (added * to the original regex), but I got a weird bug where the last search result went offscreen and added a margin to fix it

Reply:
Makes perfect sense, I like the new fix.